### PR TITLE
By default allow transactions to commit working sets with conflicts. Also fix DOLT_MERGE --abort

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
This pr introduces a new session variable that allows for transactions by default commit working sets with conflicts.

It also adds a fix for DOLT_MERGE abort which was broken before.

### CODE REVIEW

File: go/libraries/doltcore/sqle/dsess/transactions.go [hunk 1]
```diff
@@ -275,6 +258,38 @@ func (tx *DoltTransaction) doCommit(
 	return nil, nil, datas.ErrOptimisticLockFailed
 }
 
+func handleConflictSystemVariables(ctx *sql.Context, workingSet *doltdb.WorkingSet) error {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dsess/transactions.go:261]: `checkForConflictsAndConstraintViolations`

And write a method comment

File: go/libraries/doltcore/sqle/dsess/transactions.go [hunk 2]
```diff
@@ -275,6 +258,38 @@ func (tx *DoltTransaction) doCommit(
...
+	if !(allowCommitConflicts.(int8) == 1 || forceTransactionCommit.(int8) == 1) {
+		workingRoot := workingSet.WorkingRoot()
+		hasConflicts, err := workingRoot.HasConflicts(ctx)
+		if err != nil {
+			return err
+		}
+		if hasConflicts {
+			return doltdb.ErrUnresolvedConflicts
+		}
+		hasConstraintViolations, err := workingRoot.HasConstraintViolations(ctx)
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dsess/transactions.go:281]: This should actually be covered by forceTransactionCommit, but not by allowCommitConflicts

File: go/libraries/doltcore/sqle/enginetest/dolt_queries.go [hunk 3]
```diff
@@ -419,4 +422,128 @@ var DoltMerge = []enginetest.ScriptTest{
...
+				Expected: []sql.Row{{0, 1001}},
+			},
+			{
+				Query:    "SELECT DOLT_CHECKOUT('-b', 'other-branch')",
+				Expected: []sql.Row{{0}},
+			},
+		},
+	},
+	{
+		Name: "DOLT_MERGE complains when a merge overrides local changes",
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/enginetest/dolt_queries.go:529]: Should have a test where you set the new session var to false and confirm you can't commit

File: go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go [hunk 4]
```diff
@@ -792,3 +794,91 @@ var DoltTransactionTests = []enginetest.TransactionTest{
 		},
 	},
 }
+
+var DoltSqlFuncTransactionTests = []enginetest.TransactionTest{
+	{
+		Name: "Dolt Merge that commits conflicts is reflected by another user",
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go:800]: "Committed conflicts are seen by other sessions"

File: go/libraries/doltcore/sqle/dsess/transactions.go (hunk 2 above)
Comment (zachmu) [reply go/libraries/doltcore/sqle/dsess/transactions.go:281]: Also add a TODO here: you should be able to commit even with constraint violations depending on the type by setting various session vars.

For example, if you have `foreign_key_checks=0`, you should be able to commit foreign key violations, but not other kinds of constraint violations. We'll add other vars to control enforcing other kinds of checks.

And of course the `force` var should override all checks and commit no matter what

**Review by zachmu — COMMENTED** [review]
Pretty close, needs a couple changes

File: go/libraries/doltcore/sqle/enginetest/dolt_queries.go (hunk 3 above)
Comment (VinaiRachakonda) [reply go/libraries/doltcore/sqle/enginetest/dolt_queries.go:529]: Exists at line 465. Can make the test name more clear

File: go/libraries/doltcore/sqle/dsess/variables.go [hunk 5]
```diff
@@ -68,6 +69,14 @@ func init() {
 			Type:              sql.NewSystemIntType(CurrentBatchModeKey, -9223372036854775808, 9223372036854775807, false),
 			Default:           int64(0),
 		},
+		{ // If true, disables the conflict and constraint violation check when you commit a transaction.
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dsess/variables.go:72]: Comment not quite true

**Review by zachmu — APPROVED** [review]
LGTM

**Comment by andrew-wm-arthur** [comment]
Can you run the Nautobot unit tests against this one before you merge?

**Pull request merged**